elastic/kibana
# [Security Solution][Rule details] - Updates rule details page to allow tabs to be a part of url

## Summary

Similar to how the Hosts page maintains the tab state in the URL, recreating that in the rule details page. This is in anticipation of some of the exceptions UI changes being worked on where this would create a better user experience.

- Existing urls that do not contain the `tabName` param default to the `Alerts` tab (this is exactly what happens now, so no change in functionality now)
- Sample url: `/app/security/rules/id/fbfc7b80-2551-11ed-ac93-0fcc3d73cc25/rule_exceptions`
- Tab name also shows up in the breadcrumb

<img width="606" alt="Screen Shot 2022-08-26 at 1 05 08 PM" src="https://user-images.githubusercontent.com/10927944/186982927-32b7b187-22a4-49e6-9ebd-307c5bddeb86.png">

### Testing
- Navigate to a rule details url from prior to this change (or just make sure that it doesn't include a tab name) - rule should load with the `Alerts` tab active
- Select the exceptions tab. Refresh page - exceptions tab should remain active
- Breadcrumbs should be updated as you change tabs
- Selecting to filter by alert id from the execution results tab properly updates the selected tab to `Alerts` and applies expected filter

### Checklist
